File: Backend/Api/Models/Metric.cs
namespace Api.Models;

public class Metric : UpdateMetric
{
    public long Person { get; set; }

    public long User { get; set; }
}

public class UpdateMetric : MetricBase
{
    public long Id { get; set; }
}

public class CreateMetric : MetricBase
{
}

public abstract class MetricBase
{
    public DateTime Date { get; set; }

    public required string Value { get; set; }

    public string? Tag { get; set; }

    public long Type { get; set; }

    public FileTypes Source { get; set; }
}

File: flutter_frontend/lib/services/local/database/models/metric.dart
import 'package:drift/drift.dart';
import 'package:helse/services/local/database/models/helse_table.dart';
import 'package:helse/services/local/database/models/metric_type.dart';
import 'package:helse/services/local/database/models/person.dart';
import 'package:helse/services/local/database/models/unit.dart';

class Metric extends HelseTable {
  // TODO start the id at 100
  TextColumn get value => text()();
  DateTimeColumn get date => dateTime()();
  TextColumn get sourceId => text()();
  TextColumn get source => text()();
  TextColumn get tag => text().nullable()();
  late final type = integer().references(MetricType, #id)();
  late final unit = integer().nullable().references(Unit, #id)();
  late final person = integer().references(Person, #id)();
}
